File: eRents.Domain/Models/GeoRegion.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eRents.Domain.Models;

public partial class GeoRegion
{
    public int GeoRegionId { get; set; }

    [Required]
    [StringLength(100)]
    public string City { get; set; } = null!;

    [StringLength(100)]
    public string? State { get; set; }

    [Required]
    [StringLength(100)]
    public string Country { get; set; } = null!;

    [StringLength(20)]
    public string? PostalCode { get; set; }

    public virtual ICollection<AddressDetail> AddressDetails { get; set; } = new List<AddressDetail>();
}

File: e_rents_desktop/lib/models/geo_region.dart
class GeoRegion {
  final int? geoRegionId;
  final String city;
  final String? state;
  final String country;
  final String? postalCode;

  GeoRegion({
    this.geoRegionId,
    required this.city,
    this.state,
    required this.country,
    this.postalCode,
  });

  factory GeoRegion.fromJson(Map<String, dynamic> json) {
    return GeoRegion(
      geoRegionId: json['geoRegionId'] as int?,
      city: json['city'] as String? ?? '',
      state: json['state'] as String?,
      country: json['country'] as String? ?? '',
      postalCode: json['postalCode'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'geoRegionId': geoRegionId,
      'city': city,
      'state': state,
      'country': country,
      'postalCode': postalCode,
    };
  }
}
